docs(debugger): add special vars section to expression language docs (#28444)
DataDog/documentation: content/en/dynamic_instrumentation/expression-language.md

## Contextual variables

> The Expression Language provides contextual variables for different instrumentation scenarios: method instrumentation variables (`@return`, `@duration`, `@exception`) are available only when instrumenting entire methods, while collection and dictionary variables (`@it`, `@key`, `@value`) are available only within predicate expressions for filtering and transforming collections.

Reviewer: ```suggestion
The Expression Language provides contextual variables for different instrumentation scenarios: method instrumentation variables (`@return`, `@duration`, `@exception`) are available only when instrumenting entire methods, while collection and dictionary variables (`@it`, `@key`, `@value`) are only available within predicate expressions for filtering and transforming collections.
```

| Keyword     | Description                                                                |
|-------------|----------------------------------------------------------------------------|

> | `@return`   | Provides access to the return value                                        |
> | `@duration` | Provides access to the call execution duration                             |
> | `@exception`| Provides access to the current uncaught exception                          |

Reviewer 2: If you collapse the two sections into one, I'd tweak these descriptions a bit:

```suggestion
| `@return`   | Provides access to the method return value |
| `@duration` | Provides access to the method call execution duration |
| `@exception`| Provides access to the exception thrown within the method (only available if an uncaught exception exists) |
```

> | `@return`   | Provides access to the method return value |

Reviewer: ```suggestion
| `@return`   | Provides access to the method return value. |
```

> | `@duration` | Provides access to the method call execution duration |

Reviewer: ```suggestion
| `@duration` | Provides access to the method call execution duration. |
```

> | `@exception`| Provides access to the exception thrown within the method (only available if an uncaught exception exists) |

Reviewer: ```suggestion
| `@exception`| Provides access to the exception thrown within the method (only available if an uncaught exception exists). |
```

**Note**: These variables are **only available** when instrumenting an **entire method**, and **not** for individual lines of code.

> ## Special variables

Reviewer 2: `@return` etc can also be considered a "special" variable, and `@it` etc can also be considered a contextual variable. So these two headings are a bit ambiguous.

I think I'd group these two sections under a common "Special variables" heading and then have two level 3 headings called, for example, "Method instrumentation variables" and "Collection and dictionary variables".

Reviewer 3: there is also `@exception` and `@duration` :)

Author: Okay, let me collapse these two sections into a single table

## Collection operations

> When working with collections (lists, maps, etc.), you can use contextual variables in predicates to access elements during iteration. See the [Contextual variables](#contextual-variables) section for details.

Reviewer: ```suggestion
When working with collections (lists, maps, and so on), you can use contextual variables in predicates to access elements during iteration. See the [Contextual variables](#contextual-variables) section for details.
```

| `len(value_src)` | Gets the collection size. | {{< expression-language-evaluator expression="len(myCollection)" >}} |
| `[ n ]` | For collections, returns the nth item in the collection. For maps and dictionaries, returns the value that corresponds to the key `n`. If the item does not exist, the expression yields an error. | {{< expression-language-evaluator expression="myCollection[1]" >}} |

> [1]: /metrics/types/?tab=count#metric-types

Reviewer: Looks like this link is still being referenced (line 64).

Author: L64 is an argument to an expression, AFAICT these were orphaned references 🤷

Reviewer: Ah ok, thank you for clarifying!
